express-post-json/index.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-console, no-unused-vars */
+const express = require('express');
+const app = express();
+
+let nextId = 1;
+const grades = [];
+
+const parseJson = express.json();
+
+app.use(parseJson, (req, res, next) => {
+  next();
+});
+
+app.get('/api/grades', (req, res) => {
+  res.json(grades);
+});
+
+app.post('/api/grades', (req, res) => {
+  req.body.id = nextId;
+  grades.push(req.body);
+  res.status(201);
+  res.json(grades[nextId - 1]);
```

Comment:
The id and the index are not related, so don't use the id to calculate the index. you already have the object (req.body)